src/fastpath/fastpath.c

```diff
@@ -37,9 +37,11 @@ void NORETURN fastpath_vm_fault(vm_fault_type_t type)
     handler_cap = lookup_fp(TCB_PTR_CTE_PTR(NODE_STATE(ksCurThread), tcbCTable)->cap, handlerCPtr);
 #endif
 
+    /* Check that the cap is an endpoint cap and on non-mcs, that you can send to it and create the reply cap */
     if (unlikely(!cap_capType_equals(handler_cap, cap_endpoint_cap)
 #ifndef CONFIG_KERNEL_MCS
-                 || !cap_endpoint_cap_get_capCanSend(handler_cap)
+                 || !cap_endpoint_cap_get_capCanSend(handler_cap) || !(cap_endpoint_cap_get_capCanGrant(handler_cap) ||
+                                                                       cap_endpoint_cap_get_capCanGrantReply(handler_cap))
```

Comment:
I think this is easier to read as "IF there's no send rights OR (there's no grand rights and theres no grant reply rights)":
```
                  (!cap_endpoint_cap_get_capCanSend(handler_cap) || (!cap_endpoint_cap_get_capCanGrant(handler_cap) &&
                                                                       !cap_endpoint_cap_get_capCanGrantReply(handler_cap))
```

Reply:
Alright